### PR TITLE
feat: add todos_overview page

### PR DESCRIPTION
**Pull request by Limbou**
## Description

Add todos_overview page

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: test/todos_overview/view/todos_overview_page_test.dart [hunk 1]
```diff
@@ -0,0 +1,290 @@
+// ignore_for_file: prefer_const_constructors
+import 'package:bloc_test/bloc_test.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
```
Comment (orestesgaolin) [inline test/todos_overview/view/todos_overview_page_test.dart:4]:
```suggestion
import 'package:flutter/material.dart';
```

do we need cupertino explicitly?

File: test/todos_overview/view/todos_overview_page_test.dart [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        Stream.fromIterable(expectedStates),
+        initialState: TodosOverviewState(
+          status: TodosOverviewStatus.loading,
+        ),
+      );
+      await tester.pumpApp(_getView());
+      expect(find.byType(SnackBar), findsNothing);
+      await tester.pump();
+      expect(find.byType(SnackBar), findsOneWidget);
+      expect(find.text('Error occurred'), findsOneWidget);
```
Comment (orestesgaolin) [inline test/todos_overview/view/todos_overview_page_test.dart:184]: Similarly as previously referencing the l10n terms may not be safe long term. Keys or other features can be more futureproof.

File: test/todos_overview/view/todos_overview_page_test.dart (hunk 1 above)
Comment (orestesgaolin) [reply test/todos_overview/view/todos_overview_page_test.dart:4]: Oh I see, CupertinoScrollbar :)

File: lib/todos_overview/view/todos_overview_page.dart [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                            );
+                      },
+                      onDismissed: (_) {
+                        context.read<TodosOverviewBloc>().add(
+                              TodosOverviewTodoDeleted(todo),
+                            );
+                      },
+                      onTap: () {},
+                    ),
+                ],
```
Comment (orestesgaolin) [inline lib/todos_overview/view/todos_overview_page.dart:124]: What made you to choose `for` instead of `ListView.builder`?

Comment (Limbou) [reply lib/todos_overview/view/todos_overview_page.dart]: temporary drop in iq 😄

**Review by orestesgaolin — APPROVED** [review]

**Pull request merged**